### PR TITLE
Add support for new deployment endpoints

### PR DESCRIPTION
**Pull request by aron**
This adds support for the following methods:

**`deployment.list()`**

```js
await replicate.deployments.list()
{
  next: null,
  previous: null,
  results: [
    {
      owner: 'aron',
      name: 'my-app-image-generator',
      current_release: [Object]
    },
    {
      owner: 'aron',
      name: 'aron-sdxl-testing-api',
      current_release: [Object]
    },
    {
      owner: 'aron',
      name: 'arons-hello-world',
      current_release: [Object]
    },
    { owner: 'aron', name: 'aron-sdxl', current_release: [Object] }
  ]
}
```

**`deployment.create()`**

```js
await replicate.deployments.create({name: 'test-js-client', model: 'stability-ai/sdxl', version: 'da77bc59ee60423279fd632efb4795ab731d9e3ca9705ef3341091fb989b7eaf', hardware: 'gpu-a100', min_instances:0,  max_instances: 0})
{
  owner: 'aron',
  name: 'test-js-client',
  current_release: {
    number: 1,
    model: 'stability-ai/sdxl',
    version: 'da77bc59ee60423279fd632efb4795ab731d9e3ca9705ef3341091fb989b7eaf',
    created_at: '2024-03-14T11:43:32.049157Z',
    created_by: {
      type: 'user',
      username: 'aron',
      name: 'Aron Carroll',
      github_url: 'https://github.com/aron'
    },
    configuration: { hardware: 'gpu-a100', min_instances: 0, max_instances: 0 }
  }
}
```

**`deployment.update()`**

```js
await replicate.deployments.update('aron', 'my-app-image-generator', {max_instances: 0})
{
  owner: 'aron',
  name: 'my-app-image-generator',
  current_release: {
    number: 3,
    model: 'stability-ai/sdxl',
    version: 'da77bc59ee60423279fd632efb4795ab731d9e3ca9705ef3341091fb989b7eaf',
    created_at: '2024-03-14T11:42:34.934262Z',
    created_by: {
      type: 'user',
      username: 'aron',
      name: 'Aron Carroll',
      github_url: 'https://github.com/aron'
    },
    configuration: { hardware: 'gpu-t4', min_instances: 0, max_instances: 0 }
  }
}
```

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -194,6 +194,25 @@ declare module "replicate" {
...
+      update(
+        deployment_owner: string,
+        deployment_name: string,
+        deployment_config: {
+          version: string;
+          hardware: string;
+          min_instances: number;
+          max_instances: number;
+        }
+      ): Promise<Deployment>;
```
Comment (mattt) [inline index.d.ts:214]: Make config fields optional but require at least one to be specified.

```suggestion
      update(
        deployment_owner: string,
        deployment_name: string,
        deployment_config: {
          version?: string;
          hardware?: string;
          min_instances?: number;
          max_instances?: number;
        } & (
          | { version: string }
          | { hardware: string }
          | { min_instances: number }
          | { max_instances: number }
        )
      ): Promise<Deployment>;
```

File: index.d.ts [hunk 2]
```diff
@@ -33,8 +33,10 @@ declare module "replicate" {
       created_by: Account;
       configuration: {
         hardware: string;
-        min_instances: number;
-        max_instances: number;
+        scaling: {
+          min_instances: number;
+          max_instances: number;
+        };
```
Comment (mattt) [inline index.d.ts:39]: This was incorrect in the original PR adding `deployments.get` #206

File: index.d.ts (hunk 1 above)
Comment (aron) [reply index.d.ts]: Not at the moment. All fields are required on the API side for creation at the moment, but something we should fix.

Comment (aron) [reply index.d.ts]: Sorry, misread this, yes this is correct. A good fix 👍

File: index.d.ts (hunk 2 above)
Comment (mattt) [reply index.d.ts:39]: Was I out of touch? No, it was the OpenAPI specification that was wrong https://github.com/replicate/replicate-javascript/pull/232

**Pull request merged**